### ISSUE TITLE
state.generate_testcase does not use the filename provided

### ISSUE THREAD

**Issue by montyly** [trailofbits/manticore#782]
### Manticore version
Version: 0.1.7 (48f88be7590c7b15ff27cfd8c2da883d71e98190)

### Summary of the problem
`state.generate_testcase` does not use the filename provided to generate the files.
This was tested only for smart contracts.

### Step to reproduce the behavior
```
# simple.py
from manticore.ethereum import ManticoreEVM

m = ManticoreEVM() # initiate the blockchain
source_code ='''
pragma solidity^0.4.18;
contract Simple {
    function f(uint a) payable {
        if (a == 0x41) {
            throw;
        }
    }
}
'''

# Initiate the accounts
user_account = m.create_account(balance=1000)
contract_account = m.solidity_create_contract(source_code, owner=user_account, balance=0)

# Call f(a), with a symbolic value
contract_account.f(m.SValue)

# Check if an execution ends with a REVERT or INVALID
for state in m.terminated_states: 
    last_tx = state.platform.transactions[-1]
    if last_tx.result in ['REVERT','INVALID']:
        print "Error found in f() execution (see %s)"%m.workspace

        # filename should be named SimpleBug_00000000.tx
        state.generate_testcase('SimpleBug')
        break
```
`python simple.py` should generate files named "SimpleBug_*", but "test_*" is used insted

### Expected behavior
Generation of mcore_*/SimpleBug* files

### Actual behavior
Generation of mcore_*/test* files

### Any relevant logs
If I use ``m._generate_testcase_callback(state, 'SimpleBug','message')`` and this fix: https://github.com/montyly/manticore/commit/4f3a7ee059f6225bbe51275594e3c14a836a0795
 the files are correclty generated.

**Comment by disconnect3d** [comment]
@montyly hey, since you already found a fix: with such simple things, next time you can just create the Pull Request and describe the issue there :).

**Comment by ggrieco-tob** [comment]
Fixed on #791.
